### DISCUSSION TITLE
Are logs stored as files anywhere / how to export logs?

### DISCUSSION THREAD

**Discussion by ChanceHarrison** [NightscoutFoundation/xDrip#2045] [Q&A] [answered]
I would like to be able to export logs from the app so I can analyze them elsewhere. Not to mention that it would make reporting issues that request logs much easier (both for me and for the people reading them, I would imagine).

With that said, the app must store the logs somewhere, though I imagine that "somewhere" may not be accessible via a file explorer, for example. Are the logs stored in a database? Stored in some structured data format (e.g. JSON)? Plain text?

Some context around log storage would be much appreciated.

**Comment by Navid200** [comment, marked as answer]
There is a related feature request:  https://github.com/NightscoutFoundation/xDrip/issues/1864
